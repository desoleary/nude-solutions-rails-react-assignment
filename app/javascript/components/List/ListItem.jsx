import { List, Typography } from 'antd';
import { arrayOf, node, number, oneOfType, string } from 'prop-types';
import React from 'react';

const { Item } = List;
const { Text } = Typography;

Item.displayName = 'ItemAntd';

const ListItem = ({ text, extra, actions, id }) => {
  const key = `list-item-${id}`;

  return (
    <Item extra={extra} actions={actions} key={key}>
      <Text>{text}</Text>
    </Item>
  );
};

ListItem.defaultProps = {
  actions: []
};

ListItem.propTypes = {
  id: oneOfType([number, string]).isRequired,
  text: string,
  extra: node,
  actions: arrayOf(node)
};

export default ListItem;
