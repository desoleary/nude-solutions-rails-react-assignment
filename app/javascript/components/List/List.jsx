import { List as ListAntd } from 'antd';
import { arrayOf, node, number, oneOfType, shape, string } from 'prop-types';
import React from 'react';
import ListItem from './ListItem';

ListAntd.displayName = 'ListAntd';

const List = (props) => {
  const { dataSource } = props;

  return (
    <ListAntd
      dataSource={dataSource}
      size='small'
      split={false}
      renderItem={ListItem}
    />
  );
};

List.propTypes = {
  dataSource: arrayOf(
    shape({
      id: oneOfType([number, string]).isRequired,
      text: string,
      extra: node,
      actions: arrayOf(node)
    })
  )
};

export default List;
