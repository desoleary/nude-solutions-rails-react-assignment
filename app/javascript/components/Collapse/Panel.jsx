import { Collapse } from 'antd';
import { arrayOf, node, number, oneOfType, shape, string } from 'prop-types';
import React from 'react';
import List from '../List';

const { Panel: PanelAntd } = Collapse;
PanelAntd.displayName = 'PanelAntd';

const Panel = (props) => {
  const { id, header, extra, items, ...rest } = props;
  const key = `panel-${id}`;
  const listKey = `panel-list-${id}`;

  return (
    <PanelAntd {...rest} header={header} extra={extra} key={key}>
      <List dataSource={items} key={listKey} />
    </PanelAntd>
  );
};

Panel.propTypes = {
  id: oneOfType([string, number]),
  header: oneOfType([string, number, node]),
  extra: oneOfType([string, number, node]),
  items: arrayOf(
    shape({
      id: oneOfType([number, string]).isRequired,
      text: string,
      extra: node,
      actions: arrayOf(node)
    })
  )
};

export default Panel;
