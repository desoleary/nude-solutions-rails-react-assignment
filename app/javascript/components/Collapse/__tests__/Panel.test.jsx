import { Collapse } from 'antd';
import { shallow } from 'enzyme';
import React from 'react';
import Panel from '../Panel';
import List from '../../List';

const { Panel: PanelAntd } = Collapse;

describe('with shallow rendering', () => {
  let wrapper;
  const props = {
    id: 'some-identifier',
    header: <div>'some-header'</div>,
    extra: <div>some-extra-content</div>,
    items: [{ id: 'some-item-identifier' }]
  };

  beforeEach(() => {
    wrapper = shallow(<Panel {...props} />);
  });

  it('compares snapshot', () => {
    expect(wrapper.length).to.eql(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("passes props correctly to Antd's Panel", () => {
    const antdPanelWrapper = wrapper.find(PanelAntd);
    const antdPanelProps = wrapper.find(PanelAntd).props();

    expect(antdPanelWrapper.key()).to.eql(`panel-${props.id}`);
    expect(antdPanelProps.header).to.eql(props.header);
    expect(antdPanelProps.extra).to.eql(props.extra);
    expect(antdPanelProps.actions).to.eql(props.actions);
  });

  it('passes props correctly to List', () => {
    const listWrapper = wrapper.find(List);
    const listProps = listWrapper.props();

    expect(listWrapper.key()).to.eql(`panel-list-${props.id}`);
    expect(listProps.dataSource).to.eql(props.items);
  });
});
