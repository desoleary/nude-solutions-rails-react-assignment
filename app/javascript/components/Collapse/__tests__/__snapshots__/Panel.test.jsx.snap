// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with shallow rendering compares snapshot 1`] = `
<PanelAntd
  extra={
    <div>
      some-extra-content
    </div>
  }
  header={
    <div>
      'some-header'
    </div>
  }
  key="panel-some-identifier"
>
  <List
    dataSource={
      Array [
        Object {
          "id": "some-item-identifier",
        },
      ]
    }
    key="panel-list-some-identifier"
  />
</PanelAntd>
`;
